src/tools/sss_override.c

```diff
@@ -81,14 +82,21 @@ static errno_t parse_cmdline(struct sss_cmdline *cmdline,
                               &orig_name, &domain);
     if (ret != EOK) {
         ERROR("Unable to parse name %s.\n", input_name);
-        return ret;
+        goto done;
     }
 
     *_input_name = input_name;
+    if (*_input_name == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
     *_orig_name = orig_name;
     *_domain = domain;
+    ret = EXIT_SUCCESS;
 
-    return EXIT_SUCCESS;
+done:
```

Comment:
Since nothing happens in `done:` section I see it as excessive.

Keeping `return code;` would make diff shorter.

Reply:
Done.

Reply:
Still, in case `sss_tool_parse_name()` fails in `sss_override.c::parse_cmdline()` `input_name` should be freed (but not in case `sss_tool_popt_ex()` fails).

Reply:
I don't think that's necessary. `input_name` is the `input_name` member of `user` or `group` structures, depending on the caller. This member is always freed after the `parse_cmdline*` is finished. As an [example](https://github.com/SSSD/sssd/pull/6350/files#diff-fb2d8709a9a434b762ee1c9077d7f881185652bffdcd995db1e187ee3b2e02f5R1437):
```
    ret = parse_cmdline_user_add(cmdline, tool_ctx, &user);
    if (ret != EOK) {
        DEBUG(SSSDBG_CRIT_FAILURE, "Unable to parse command line.\n");
        goto done;
    }
...
done:
    free(discard_const(user.input_name));

    return ret;
```

Reply:
`input_name` is a local variable of `parse_cmdline()` function.
It points to a memory allocated via `sss_tool_popt_ex()`.
In case `sss_tool_parse_name()` fails and `parse_cmdline()` returns, this memory leaks.

Reply:
You are right. I changed that part of the code.